### PR TITLE
Add MRAP support for CRT s3transfer integration

### PR DESCRIPTION
**Pull request by nateprewitt**
This PR (previously #9076) patches a gap with SigV4a signing for the CRT S3 transfer manager. On performance-optimized instance types [[list](https://awscli.amazonaws.com/v2/documentation/api/latest/topic/s3-config.html#preferred-transfer-client)], the CRT will auto-enable itself for all high-level `aws s3` commands. This may lead to signature mismatches for [Multi-Region Access Points](https://docs.aws.amazon.com/AmazonS3/latest/userguide/MultiRegionAccessPointRequests.html) backed by buckets in other regions.

Customers currently encountering this issue can bypass the problem by setting `preferred_transfer_client` to `classic` ([docs](https://awscli.amazonaws.com/v2/documentation/api/latest/topic/s3-config.html#preferred-transfer-client)). That mitigation will no longer be needed after this patch is merged.

### CODE REVIEW

File: tests/functional/s3transfer/test_crt.py [hunk 1]
```diff
@@ -156,6 +160,26 @@ def _assert_exected_s3express_request(
             awscrt.auth.AwsSigningAlgorithm.V4_S3EXPRESS
         )
 
+    def _assert_exected_mrap_request(
```
Comment (kdaily) [inline tests/functional/s3transfer/test_crt.py:163]: nit:

```suggestion
    def _assert_expected_mrap_request(
```

This is a copy paste from other mis-named tests, so fine to clean this up at another time.

Comment (nateprewitt) [reply tests/functional/s3transfer/test_crt.py:163]: I'll do a quick follow up PR cleaning up everything so we don't lose it if one of these needs to get reverted.

**Comment by kdaily** [comment]
LGTM. Tested on a `trn1.32xlarge` and that worked as expected.

**Review by kdaily — APPROVED** [review]

File: tests/functional/s3transfer/test_crt.py (hunk 1 above)
Comment (nateprewitt) [reply tests/functional/s3transfer/test_crt.py:163]: https://github.com/aws/aws-cli/pull/9087

**Pull request merged**